dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/src/main/java/org/apache/dolphinscheduler/plugin/registry/etcd/EtcdRegistry.java

```diff
@@ -315,27 +312,41 @@ public boolean acquireLock(String key) {
             // keep the lease
             client.getLeaseClient().keepAlive(leaseId, Observers.observer(response -> {
             }));
-            lockClient.lock(byteSequence(key), leaseId).get();
+            lockClient.lock(byteSequence(lockKey), leaseId).get();
 
             // save the leaseId for release Lock
-            leaseIdMap.put(key, leaseId);
+            lockMap.put(lockKey, new LockEntry(leaseId));
             return true;
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             throw new RegistryException("etcd get lock error", e);
         } catch (Exception e) {
-            throw new RegistryException("etcd get lock error, lockKey: " + key, e);
+            throw new RegistryException("etcd get lock error, lockKey: " + lockKey, e);
+        }
+    }
+
+    private static boolean currentThreadIsReentrant(String lockKey, Map<String, LockEntry> lockEntryMap) {
+        LockEntry lockEntry = lockEntryMap.get(lockKey);
+        if (lockEntry != null) {
+            lockEntry.lockCount.incrementAndGet();
+            return true;
         }
+        return false;
+    }
```

Comment:
The method name is strange.
```suggestion
    private static boolean acquireBasedOnThreadHeldLocks(String lockKey, Map<String, LockEntry> threadHeldLocks) {
        LockEntry lockEntry = threadHeldLocks.get(lockKey);
        if (lockEntry != null) {
            lockEntry.lockCount.incrementAndGet();
            return true;
}
        return false;
    }
```